### PR TITLE
[SNOWFLAKE] Add missing function

### PR DESCRIPTION
**Pull request by MasterKuat**
predicate could be a select item
add trim familly function
add like, ilike and rlike predicate

### CODE REVIEW

File: sql/snowflake/SnowflakeParser.g4 [hunk 1]
```diff
@@ -3889,7 +3900,9 @@ predicate
     | expr comparison_operator (ALL | SOME | ANY) '(' subquery ')'
     | expr NOT* BETWEEN expr AND expr
     | expr NOT* IN '(' (subquery | expr_list) ')'
```
Comment (hellozrh) [inline sql/snowflake/SnowflakeParser.g4:3902]: the keyword NOT should only be used one or zero times. So :
```g4
    | expr NOT? BETWEEN expr AND expr
    | expr NOT? IN '(' (subquery | expr_list) ')'
```

File: sql/snowflake/SnowflakeParser.g4 [hunk 2]
```diff
@@ -3641,6 +3651,7 @@ as_alias
 
 expression_elem
     : expr as_alias?
+    | predicate as_alias?
```
Comment (hellozrh) [inline sql/snowflake/SnowflakeParser.g4:3654]:
```g4
    : (expr | predicate) as_alias?
```

Comment (MasterKuat) [reply sql/snowflake/SnowflakeParser.g4]: Fixed, thanks

File: sql/snowflake/SnowflakeParser.g4 (hunk 1 above)
Comment (MasterKuat) [reply sql/snowflake/SnowflakeParser.g4]: Wasn't mine but sure fixed.

File: sql/snowflake/SnowflakeParser.g4 [hunk 3]
```diff
@@ -3331,6 +3335,7 @@ binary_or_ternary_builtin_function
     : CHARINDEX
     | REPLACE
     | substring=( SUBSTRING | SUBSTR )
+    | ( LIKE | ILIKE )
```
[the remark this answers is not in the document]
Comment (MasterKuat) [reply sql/snowflake/SnowflakeParser.g4]: Fixed

**Pull request merged**